pex/bin/pex.py

```diff
@@ -647,8 +658,12 @@ def walk_and_do(fn, src_dir):
       for resolved_dist in resolveds:
         log('  %s -> %s' % (resolved_dist.requirement, resolved_dist.distribution),
             V=options.verbosity)
-        pex_builder.add_distribution(resolved_dist.distribution)
-        pex_builder.add_requirement(resolved_dist.requirement)
+
+        if options.dehydrated:
+          pex_info.add_dehydrated_requirement(resolved_dist.distribution.as_requirement())
```

Comment:
This is incorrect. `Distribution.as_requirement()` loses environment markers, you must use `resolved_dist.requirement`.